src/components/_addProjectsModal/AddProjectsModal.js

```diff
@@ -0,0 +1,162 @@
+import Modal from "../modal/Modal";
+import { useState, useEffect } from "react";
+import FormFields from "./utils/FormFields";
+import FormField from "./utils/FormField";
+import ProjectStatus from "./utils/ProjectStatus";
+
+const AddProjectsModal = ({ handleClose, isOpen, isLoading, actions, employees }) => {
+  const [name, setName] = useState("");
+  const [description, setDescription] = useState("");
+  const [duration, setDuration] = useState({
+    from: "",
+    to: "",
+  });
+  const [developers, setDevelopers] = useState([]);
+  const [hourlyRate, setHourlyRate] = useState(0);
+  const [projectValue, setProjectValue] = useState(0);
+  const [projectStatus, setProjectStatus] = useState(ProjectStatus.Active);
+  const [developerOptions, setDeveloperOptions] = useState([]);
+
+  useEffect(() => {
+    fetchDevelopers();
+  }, []);
+
+  const resetState = () => {
+    setName("");
+    setDescription("");
+    setDuration({
+      from: "",
+      to: "",
+    });
+    setDevelopers([]);
+    setHourlyRate(0);
+    setProjectValue(0);
+    setProjectStatus(ProjectStatus.Active);
+  };
+
+  const fetchDevelopers = async () => {
+    try {
+      actions.fetchAllEmployeesRequest();
+    } catch (error) {
+      console.error("Error fetching developers:", error);
+    }
+  };
+
+  useEffect(() => {
+    setDeveloperOptions(
+      employees.map((developer) => ({
+        value: developer._id,
+        label: `${developer.firstName} ${developer.lastName}`,
+        employee: developer,
+      }))
+    );
+  }, [employees]);
+
+  const handleSubmit = async () => {
+    try {
+      const formattedDevelopers = developers.map((developerId) => {
```

Comment:
I think I already see this block of code when you fetch developers for project table. If I am right then you can make helper function for this

Reply:
and also try to avoid loop in loop because now you have n * n complexity. Sometimes https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce this can fix complexity